tests/testsuite/pub_priv.rs

```diff
@@ -479,3 +479,220 @@ fn allow_priv_in_custom_build() {
         )
         .run()
 }
+
+// A indirectly add D as private dependency.
```

Comment:
These tests don't make it too clear that they are not expected behavior but that some help show existing bugs.  We should call that out, linking to the issue in question